http3/client.go

```diff
@@ -74,14 +75,15 @@ func (c *SingleDestinationRoundTripper) Start() Connection {
 }
 
 func (c *SingleDestinationRoundTripper) init() {
-	c.logger = utils.DefaultLogger.WithPrefix("h3 client")
-	c.requestWriter = newRequestWriter(c.logger)
+	c.requestWriter = newRequestWriter()
 	c.decoder = qpack.NewDecoder(func(hf qpack.HeaderField) {})
-	c.hconn = newConnection(c.Connection, c.EnableDatagrams, c.UniStreamHijacker, protocol.PerspectiveClient, c.logger)
+	c.hconn = newConnection(c.Connection, c.EnableDatagrams, c.UniStreamHijacker, protocol.PerspectiveClient, c.Logger)
 	// send the SETTINGs frame, using 0-RTT data, if possible
 	go func() {
 		if err := c.setupConn(c.Connection); err != nil {
-			c.logger.Debugf("Setting up connection failed: %s", err)
+			if c.Logger != nil {
+				c.Logger.Debug("setting up connection failed", "error", err)
+			}
```

Comment:
Question: should we always fallback to `slog.Default()` if no logger is explicitly set?

Reply:
I'm not sure. I don't want to spam peoples' logs with debug messages, so I'm inclined to leave it as is far now. That said, I don't think we have a good / consistent logging philosophy at the moment, so I'd be happy to discuss and revise this decision in the future.